### DISCUSSION TITLE
Auto closing brackets, quotes and parenthesis in a text input

### DISCUSSION THREAD

**Discussion by ricardomatias** [react-hook-form/react-hook-form#6005] [Q&A]
How can I auto close brackets, quotes and parenthesis automatically before the input is validated? If the user writes `hello "` it should append another `"` at the end of the input. This value should be changed everywhere.

**Comment by bluebill1049** [comment]
Sounds like a mask input component.

**Reply by ricardomatias** [reply]
I wasn't clear enough, my question is regarding the react-hook-form part, where can I intercept the `value` before it's propagated to validation and the form data.The symbols duplication is not an issue.
